lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -33,7 +34,45 @@ def overage(issue)
           end
 
           def process_file(file)
-            RubyParser.new.process(File.binread(file), file, TIMEOUT)
+            if File.extname(file) == ".erb"
+              process_erb file
+            else
+              RubyParser.new.process(File.binread(file), file, TIMEOUT)
+            end
+          end
+
+          # taken from Flay, https://github.com/seattlerb/flay/blob/master/lib/flay_erb.rb
+
+          def process_erb(file)
+            erb = File.read(file)
+
+            ruby = Erubis.new(erb).src
+            begin
+              RubyParser.new.process(ruby, file)
+            rescue => e
+              $stderr.puts ruby
```

Comment:
This seems a bit excessive, and could be very large.

We catch exceptions up the stack, and we're raising right below here anyway: we can probably lose the begin/rescue here. We'll get the filename up the stack & us/customers can user that for debugging.